File: 1813-maximum-erasure-value/1813-maximum-erasure-value.py
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        n=len(nums)
        seen=set()
        currsum=0
        maxsum=float("-inf")
        l,r=0,0
        while(r<n):
            while(nums[r] in seen):
                seen.remove(nums[l])
                currsum-=nums[l]
                l+=1
            seen.add(nums[r])
            currsum+=nums[r]
            maxsum=max(maxsum,currsum)
            r+=1
        return maxsum        

        

File: 1813-maximum-erasure-value/1813-maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        unordered_set<int> seen;
        int maxSum = 0, currentSum = 0;
        int left = 0;

        for (int right = 0; right < nums.size(); ++right) {
            while (seen.count(nums[right])) {
                // Remove from current window and set
                seen.erase(nums[left]);
                currentSum -= nums[left];
                ++left;
            }
            // Add current number
            seen.insert(nums[right]);
            currentSum += nums[right];
            maxSum = max(maxSum, currentSum);
        }

        return maxSum;
    }
};